src/grammar/mod.rs

```diff
@@ -216,6 +217,49 @@ impl Scenario {
             })
             .collect()
     }
+
+    pub(crate) fn validate(&self) -> Result<()> {
+        let expressions = self
+            .expressions
+            .iter()
+            .filter(|(id, _)| id.0 != "absent")
+            .map(|(_, expr)| expr.normalize2(self))
+            .collect::<Result<HashSet<_>>>()?;
```

Comment:
Usually it is easier on the eye to put the type behind the let.

Reply:
Yes but that won't work in this case, since I'm collapsing/flattening the `Result<Formula>` items of the Iterator (such that there's only one result at the end) while using the `?` operator at the same time.

Reply:
Have tried different constellations, this is the one with the fewest characters and maintaining okay-ish readability